### PR TITLE
Partial service deployments

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR removes the need for network constants

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have locally run services that could be impacted and they do not present failures derived from my changes

### CODE REVIEW

File: packages/valory/services/apy_estimation_hardhat/service.yaml [hunk 1]
```diff
@@ -8,7 +8,12 @@ fingerprint:
   README.md: bafybeibqyfur7wusrbfga66hcbauxvhs2g3j5yhknsxlvyh6mil57aajyu
 fingerprint_ignore_patterns: []
 agent: valory/apy_estimation:0.1.0:bafybeieenxibmafnljvcwapwaw4kaktbgzbtoaz3lzbd44wbfx3gkgpoj4
-network: hardhat
+network:
+  id: hardhat
+  address: http://localhost:8545
+  chain_id: 31337
+  poa_chain: false
+  default_gas_price_strategy: eip1559
```
Comment (8ball030) [inline packages/valory/services/apy_estimation_hardhat/service.yaml:16]: I like this a lot more

**Review by 8ball030 — DISMISSED** [review]
LGTM
I assume you have run this?

File: autonomy/configurations/base.py [hunk 2]
```diff
@@ -134,7 +135,11 @@ def __init__(  # pylint: disable=too-many-arguments
         self.agent = PublicId.from_str(str(agent))
         self.description = description
         self.number_of_agents = number_of_agents
-        self.network = network
+
+        if network is None:
+            self.network = DEFAULT_NETWORK_CONFIG
+        else:
+            self.network = network
```
Comment (dvilelaf) [inline autonomy/configurations/base.py:142]:
```suggestion
       self.network = network or DEFAULT_NETWORK_CONFIG
```

**Comment by 8ball030** [comment]
image

@angrybayblade what is this? Looks like a new failure on windows?

**Comment by Adamantios** [comment]
> image
> 
> @angrybayblade what is this? Looks like a new failure on windows?

This is a network issue. If you look a bit before:

```
Failures
 - wget (exited 1) - wget not installed. An error occurred during installation:
 The remote server returned an error: (503) Server Unavailable. Service Unavailable
```

File: packages/valory/services/counter/service.yaml [hunk 3]
```diff
@@ -9,4 +9,9 @@ fingerprint:
 fingerprint_ignore_patterns: []
 number_of_agents: 1
 agent: valory/counter:0.1.0:bafybeidmoafileokvz4wrx2o4ry2sgtpngcvdn63pkmqswm5qjctnes6b4
-network: hardhat
+network:
+  id: hardhat
+  address: http://localhost:8545
+  chain_id: 31337
+  poa_chain: false
+  default_gas_price_strategy: eip1559
```
Comment (DavidMinarsch) [inline packages/valory/services/counter/service.yaml:17]: These are already defined on the ledger connection, I am not sure why we are redefining this here...

Comment (angrybayblade) [reply packages/valory/services/counter/service.yaml:17]: These are service level overrides, these variables will be exported as env variables in a docker container and picked up by the aea framework when building the aea.

Comment (DavidMinarsch) [reply packages/valory/services/counter/service.yaml:17]: Then they should be specified as overrides for the underlying component on the relevant yaml page, same as we do for the agents imo.

File: packages/valory/services/oracle_ropsten_cluster/service.yaml [hunk 4]
```diff
@@ -102,3 +96,13 @@ models:
       args: *id001
   - server_api:
       args: *id002
+---
+public_id: valory/abci:0.1.0
```
Comment (DavidMinarsch) [inline packages/valory/services/oracle_ropsten_cluster/service.yaml:100]: It's the `valory/ledger` connection not the `valory/abci` one

File: docs/quick_start.md [hunk 5]
```diff
@@ -51,23 +51,13 @@ export OPEN_AEA_IPFS_ADDR="/dns/registry.autonolas.tech/tcp/443/https"
 touch Pipfile && pipenv --python 3.10 && pipenv shell
```
Comment (DavidMinarsch) [inline docs/quick_start.md:51]: Please keep changes here!

Comment (angrybayblade) [reply docs/quick_start.md:51]: I'm confused, what changes?

File: docs/quick_start.md [hunk 6]
```diff
@@ -51,23 +51,13 @@ export OPEN_AEA_IPFS_ADDR="/dns/registry.autonolas.tech/tcp/443/https"
...
-autonomy init --remote
-```
-
 ## Deploy a Local Agent Service
 
-!!! note
-    On **MacOS** and **Windows**, running Docker containers requires having Docker Desktop running as well. If you're using one of those systems, remember to start Docker Desktop
-    before you run agent services.
-
-
```
Comment (dvilelaf) [inline docs/quick_start.md:70]: Why are we removing this from the quickstart?

File: docs/autonomy.md [hunk 7]
```diff
@@ -306,62 +306,3 @@ autonomy analyse benchmarks abci_build/persistent_data/benchmarks
...
 
 By default tool will generate output for all periods but you can specify which period to generate output for, same goes for block types as well.
-
-
-## Fetch
-
-```bash
-Usage: autonomy fetch [OPTIONS] PUBLIC_ID_OR_HASH
-
-  Fetch an agent from the registry.
```
Comment (dvilelaf) [inline docs/autonomy.md:316]: Why removed?

File: docs/quick_start.md (hunk 6 above)
Comment (angrybayblade) [reply docs/quick_start.md:70]: Looks like the merge errors, I'll revert them

File: docs/autonomy.md (hunk 7 above)
Comment (angrybayblade) [reply docs/autonomy.md:316]: Same as above

**Review by Karrenbelt — COMMENTED** [review]
https://github.com/valory-xyz/open-autonomy/blob/2b128ccc836089fde28f55ae5bca60169652c4d7/tests/helpers/deployments/test_deployments.py#L237-L250
(why) is checking absence of "hardhat" a good way to assess the creation of Ropsten deployment?


https://github.com/valory-xyz/open-autonomy/blob/2b128ccc836089fde28f55ae5bca60169652c4d7/tests/helpers/deployments/test_deployments.py#L256-L260
complete test or remove

https://github.com/valory-xyz/open-autonomy/blob/2b128ccc836089fde28f55ae5bca60169652c4d7/tests/helpers/deployments/test_deployments.py#L335
this is odd. Should this be `with pytest.raises()`? If so please also add `match="<some partial error message>"`

**Review by 8ball030 — COMMENTED** [review]
LGTM
My only concern is that we now need a different service spec for deploying to a hardhat cluster if I am not mistaken.

**Review by dvilelaf — COMMENTED** [review]
Looking good. Let's wait for tests.

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**